datumaro/components/extractor.py

```diff
@@ -517,9 +581,40 @@ class DatasetItem:
         type=str, validator=not_empty)
     annotations = attrib(factory=list, validator=default_if_none(list))
     subset = attrib(converter=lambda v: v or DEFAULT_SUBSET_NAME, default=None)
+
+    # Currently unused
     path = attrib(factory=list, validator=default_if_none(list))
 
+    # TODO: introduce "media" field with type info. Replace image and pcd.
     image = attrib(type=Image, default=None)
+    pcd = attrib(type=bytes, default=None)
+    related_images = attrib(type=list, default=None)
+
+    def __attrs_post_init__(self):
+        assert not (self.has_image and self.has_pcd), \
+            "Can't set both image and point cloud info"
+
+    @related_images.validator
+    def _related_image_validator(self, attribute, related_images):
```

Comment:
This looks more like a converter than a validator.

Reply:
You're right, but converters have lesser capabilities than validators, so I decided to avoid using converters in general.

Reply:
Refactored